### PR TITLE
handling wrong batch destination config case for batch router

### PR DESCRIPTION
**Pull request by kuldeep0020**
**Fixes** # (*issue*)
https://www.notion.so/rudderstacks/Don-t-panic-on-dest-miss-config-caa5ae4b225343f5ba6ccc8c47e050f9

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: router/batchrouter/batchrouter.go [hunk 1]
```diff
@@ -969,16 +975,23 @@ func (brt *HandleT) setJobStatus(batchJobs *BatchJobsT, isWarehouse bool, err er
...
-		batchReqMetric.batchRequestFailed = 1
-		// We keep track of number of failed attempts in case of failure and number of events uploaded in case of success in stats
+	if err != nil {
+		switch err.Error() {
+		case DisabledEgress:
+			brt.logger.Debugf("BRT: Outgoing traffic disabled : %v at %v", batchJobs.BatchDestination.Source.ID, time.Now().Format("01-02-2006"))
+			batchJobState = jobsdb.Succeeded.State
+			errorResp = []byte(fmt.Sprintf(`{"success":"%s"}`, DisabledEgress))
+		case InvalidServiceProvider:
+			brt.logger.Debugf("BRT: Destination service provider not supported : %v at %v", batchJobs.BatchDestination.Source.ID, time.Now().Format("01-02-2006"))
```
Comment (chandumlg) [inline router/batchrouter/batchrouter.go:985]: Log destination id and destination type as well.

File: services/filemanager/filemanager.go [hunk 2]
```diff
@@ -14,6 +14,9 @@ var (
 	DefaultFileManagerFactory FileManagerFactory
 )
 
+const (
+	InvalidServiceProvider = "service provider not supported"
```
Comment (chandumlg) [inline services/filemanager/filemanager.go:18]: Make common error package and use it everywhere.

Comment (cisse21) [reply services/filemanager/filemanager.go:18]: Probably a more descriptive error message...  If its an empty string it's not configured else it isn't supported?

**Review by cisse21 — CHANGES_REQUESTED** [review]

File: router/batchrouter/batchrouter.go (hunk 1 above)
Comment (kuldeep0020) [reply router/batchrouter/batchrouter.go:985]: done

File: services/filemanager/filemanager.go (hunk 2 above)
Comment (kuldeep0020) [reply services/filemanager/filemanager.go:18]: done

File: router/batchrouter/batchrouter.go [hunk 3]
```diff
@@ -1030,14 +1043,15 @@ func (brt *HandleT) setJobStatus(batchJobs *BatchJobsT, isWarehouse bool, err er
...
-			// change job state to abort state after warehouse service is continuously failing more than warehouseServiceMaxRetryTimeinHr time
-			if jobState == jobsdb.Failed.State && isWarehouse && postToWarehouseErr {
+		switch jobState {
+		case jobsdb.Failed.State:
+			if timeElapsed > brt.retryTimeWindow && job.LastJobStatus.AttemptNum >= brt.maxFailedCountForJob && !postToWarehouseErr {
+				job.Parameters = misc.UpdateJSONWithNewKeyVal(job.Parameters, "stage", "batch_router")
+				abortedEvents = append(abortedEvents, job)
+				router.PrepareJobRunIdAbortedEventsMap(job.Parameters, jobRunIDAbortedEventsMap)
+				jobState = jobsdb.Aborted.State
+			} else if isWarehouse && postToWarehouseErr {
```
[the remark this answers is not in the document]
Comment (kuldeep0020) [reply router/batchrouter/batchrouter.go:1053]: done

File: services/filemanager/filemanager.go [hunk 4]
```diff
@@ -80,7 +83,7 @@ func (factory *FileManagerFactoryT) New(settings *SettingsT) (FileManager, error
 			Config: GetDOSpacesConfig(settings.Config),
 		}, nil
 	}
-	return nil, errors.New("No provider configured for FileManager")
+	return nil, errors.New(InvalidServiceProvider)
```
[the remark this answers is not in the document]
Comment (kuldeep0020) [reply services/filemanager/filemanager.go:86]: done

**Review by chandumlg — APPROVED** [review]

**Review by cisse21 — APPROVED** [review]

**Review by lvrach — APPROVED** [review]

**Pull request merged**